src/config.js

```diff
@@ -1,20 +1,120 @@
-let config = {
-  senderEmail: '',
-  senderPassword: '',
-  hostingURL: '',
-}
+const validator = require('validator').default;
+const mailer = require('nodemailer');
+
+/**
+ * @typedef {Object} config
+ * @property  {URL} hostingURL
+ * @property  {"Custom"|"126"|"163"|"1und1"|"AOL"|"DebugMail"|"DynectEmail"|"FastMail"|"GandiMail"|"Gmail"|"Godaddy"|"GodaddyAsia"|"GodaddyEurope"|"hot.ee"|"Hotmail"|"iCloud"|"mail.ee"|"Mail.ru"|"Maildev"|"Mailgun"|"Mailjet"|"Mailosaur"|"Mandrill"|"Naver"|"OpenMailBox"|"Outlook365"|"Postmark"|"QQ"|"QQex"|"SendCloud"|"SendGrid"|"SendinBlue"|"SendPulse"|"SES"|"SES-US-EAST-1"|"SES-US-WEST-2"|"SES-EU-WEST-1"|"Sparkpost"|"Yahoo"|"Yandex"|"Zoho"|"qiye.aliyun"} mailService - Set mailing service to use defaults to gmail
+ * @property  {String} senderEmail set sender's email or user only in case of supported services
+ * @property  {String} senderPassword set sender's password only in case of supported services
+ * @property  {Object} [auth]
+ * @property  {String} auth.senderEmail set sender's email or user only in case of custom service
+ * @property  {String} auth.senderPassword set sender's password only in case of custom service
+ * @property  {String} [host] - Host for your custom service
+ * @property  {Number} [port] - Port for your custom service
+ * @property  {Boolean} [secure] - Set to true for your custom service if you want a (TLS) secured connection
+ */
 
-const setConfig = (userConfig) => {
-  // TODO: Better  validation with joi
-  if (userConfig.senderEmail && userConfig.senderPassword) {
-    config = userConfig
+/**
+ * @private
+ * @type {config} configuration for mailer service
+ */
+let config = {}
+/**
+ * @exports 
+ * @param  {Object} userConfig set sender's email configuration
+ * @param  {URL} userConfig.hostingURL
+ * @param  {"Custom"|"126"|"163"|"1und1"|"AOL"|"DebugMail"|"DynectEmail"|"FastMail"|"GandiMail"|"Gmail"|"Godaddy"|"GodaddyAsia"|"GodaddyEurope"|"hot.ee"|"Hotmail"|"iCloud"|"mail.ee"|"Mail.ru"|"Maildev"|"Mailgun"|"Mailjet"|"Mailosaur"|"Mandrill"|"Naver"|"OpenMailBox"|"Outlook365"|"Postmark"|"QQ"|"QQex"|"SendCloud"|"SendGrid"|"SendinBlue"|"SendPulse"|"SES"|"SES-US-EAST-1"|"SES-US-WEST-2"|"SES-EU-WEST-1"|"Sparkpost"|"Yahoo"|"Yandex"|"Zoho"|"qiye.aliyun"} userConfig.mailService- Set mailing service to use defaults to gmail
```

Comment:
Let's limit the number of officially supported services to gmail and ses as discussed in #7

Reply:
Hey,
There's a misunderstanding here, it's not like we have to support those, those are already supported by node mailer, we are just letting the user know that it is supported. These options are there in the node mailer for a long time. So there is no reason to worry about it.